.github/workflows/e2e-test-heroku.yml

```diff
@@ -57,3 +57,42 @@ jobs:
           CONNECT_MANAGEMENT_URL: ${{ secrets.CONNECT_MANAGEMENT_URL }}
           CONNECT_MANAGEMENT_API_KEY: ${{ secrets.CONNECT_MANAGEMENT_API_KEY }}
         run: yarn e2e:remove-test-user
+  
+  e2e-tests-check-status: 
+    if: ${{ contains(github.event.deployment_status.target_url, 'heroku') && always() }}
+    needs: e2e-tests-heroku
+    runs-on: ubuntu-latest
+
+    container:
+      image: node:14.15.1
+
+    env:
+      GITHUB_CONTEXT_EVENT: ${{ toJson(github.event) }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: |
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+            **/node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: dependencies
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install
```

Comment:
@Fenntasy I was wondering myself if we really need to checkout the project and install every deps, since in this script (`update-e2e-run-check-status.ts`), we don't use many of the project deps. Couldn't we just `yarn install ...` the deps we need for this script execution and save some time on this job ?

Reply:
If `yarn` (or `npm`) worked in the way you think, yes we could absolutely do that.
Unfortunately, `yarn install package` will install everything:

```sh
~/W/f/connect-account ❯ rm -rf node_modules
~/W/f/connect-account ❯ yarn add jest
yarn add v1.22.5
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
[4/5] 🔗  Linking dependencies...
[5/5] 🔨  Building fresh packages...
success Saved lockfile.
warning "jest" is already in "devDependencies". Please remove existing entry first before adding it to "dependencies".
success Saved 1 new dependency.
info Direct dependencies
└─ jest@26.6.3
info All dependencies
└─ jest@26.6.3
✨  Done in 35.94s.
~/W/f/connect-account ❯ yarn
yarn install v1.22.5
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
success Already up-to-date.
✨  Done in 0.57s.
```

Reply:
I see, but what if if we don't execute this step inside the project ? (I mean, without that `- uses: actions/checkout@v2`). There won't be any `package.json` so we possibly could install just what we need, no ? 🤔  
(I'm not saying that we absolutely need to do that, but it's a question that I had when working on this task and I wanted your thought about it)

Reply:
How would you get the script then?

Reply:
Yep, just after writing the last comment, I realised that. Forget about what I said 😅

Reply:
I didn't mean it like that, it would be totally possible to fetch only the script we need (via github blog urls) for instance.
But that would be harder to deal with when we're updating the script.

Reply:
Thinking out loud, one way of doing that would be to have another `package.json` in the `bin` folder and consider it totally separate from the rest of the application (that would force us to not use the rest of the code though). But it would be a considerably smaller installation

Reply:
Hmm I don't really see how to do that. Does this action `actions/checkout@v2` have some options to checkout more "deeper" in the project ? Or would it be more like giving volumes to a docker image ?

Reply:
Well, first, it's not something I would suggest to do in this PR but in the future.

But no, I don't know if we can make a deep checkout but that action is around 2 seconds I think so that's really not the problem. The long part is the get-cache/yarn-install/set-cache and that could be done faster

Reply:
Yep, I totally agree, it's not something worth doing now. But I would like to discuss it one day, if we decide to go that way 🙂